aten/src/ATen/core/function_schema_inl.h

```diff
@@ -151,6 +151,10 @@ inline void FunctionSchema::checkArg(
     const IValue& value,
     const Argument& argument,
     optional<size_t> pos) const {
+  if (value.isTensor() && argument.type() == TensorType::get()) {
```

Comment:
This is correct (and faster, yay!) but still does 2 shared_ptr operations (argument.type() and TypeTensor::get() both return shared_ptr<Type> which has to be destructed).
`argument.type()->kind() == TensorType::kind()` would eliminate 1 shared_ptr creation. Modifying argument.type() to return a `const TypePtr&` would eliminate the other. Not sure if they will speed things up further, but worth checking.

Reply:
I thought about checking only the type kind, but wasn't sure it would correctly work in a theoretical case where argument type would have, say, shapes specialized. I.e. if the graph would look like:
```
graph(%input : Float(10))
```
In that case, IIUC, `argument.type()->kind()` would equal `TensorType::kind()`, but `argument.type()` would not be equal `TensorType::get()`.

As for your second suggestion, let me try that. FWIW, I also tried replacing signature of `Type::isSubtypeOfExt` to use a reference for the first argument, but I didn't notice improved performance from that.

Reply:
Submitted #48061 for that, PTAL @zdevito!